File: POLIMORFISMO/EJERCICIO-9/JAVA/Minecraft.java
public class Minecraft {
    public static void main(String[] args) {
        BloqueCofre bloque1 = new BloqueCofre(30, 10, "madera");
        BloqueCofre bloque2 = new BloqueCofre(50, 20, "hierro");

        BloqueTnt bloque3 = new BloqueTnt("básica", 100);
        BloqueTnt bloque4 = new BloqueTnt("potente", 200);

        BloqueHorno bloque5 = new BloqueHorno("gris", 5);
        BloqueHorno bloque6 = new BloqueHorno("negro", 10);

        Object[] bloques = {bloque1, bloque2, bloque3, bloque4, bloque5, bloque6};

        for (Object bloque : bloques) {
            if (bloque instanceof BloqueCofre) {
                ((BloqueCofre) bloque).accion();
                ((BloqueCofre) bloque).colocar("suelo");
                ((BloqueCofre) bloque).romper();
            } else if (bloque instanceof BloqueTnt) {
                ((BloqueTnt) bloque).accion();
                ((BloqueTnt) bloque).colocar("suelo");
                ((BloqueTnt) bloque).romper();
            } else if (bloque instanceof BloqueHorno) {
                ((BloqueHorno) bloque).accion();
                ((BloqueHorno) bloque).colocar("suelo");
                ((BloqueHorno) bloque).romper();
            }
            System.out.println("------------------------");
        }
    }
}

class BloqueCofre {
    int capacidad;
    int resistencia;
    String tipo;

    public BloqueCofre(int capacidad, int resistencia, String tipo) {
        this.capacidad = capacidad;
        this.resistencia = resistencia;
        this.tipo = tipo;
    }

    public void accion() {
        System.out.println("Este cofre puede almacenar objetos.");
    }

    public void colocar(String orientacion) {
        System.out.println("Cofre colocado en: " + orientacion);
    }

    public void romper() {
        System.out.println("Has roto un cofre, los objetos salen volando.");
    }
}

class BloqueTnt {
    String tipo;
    int dano;

    public BloqueTnt(String tipo, int dano) {
        this.tipo = tipo;
        this.dano = dano;
    }

    public void accion() {
        System.out.println("La TNT está lista para explotar.");
    }

    public void colocar(String orientacion) {
        System.out.println("TNT colocada en: " + orientacion);
    }

    public void romper() {
        System.out.println("La TNT se activa y explota.");
    }
}

class BloqueHorno {
    String color;
    int capacidadComida;

    public BloqueHorno(String color, int capacidadComida) {
        this.color = color;
        this.capacidadComida = capacidadComida;
    }

    public void accion() {
        System.out.println("Este horno puede cocinar comida.");
    }

    public void colocar(String orientacion) {
        System.out.println("Horno colocado en: " + orientacion);
    }

    public void romper() {
        System.out.println("El horno se rompe y suelta los objetos cocinados.");
    }
}

File: POLIMORFISMO/EJERCICIO-9/PYTHON/Minecraft.py
class BloqueCofre:
    def __init__(self, capacidad, resistencia, tipo):
        self.capacidad = capacidad
        self.resistencia = resistencia
        self.tipo = tipo

    def accion(self):
        print("Este cofre puede almacenar objetos.")

    def colocar(self, orientacion):
        print("Cofre colocado en:", orientacion)

    def romper(self):
        print("Has roto un cofre, los objetos salen volando.")


class BloqueTnt:
    def __init__(self, tipo, dano):
        self.tipo = tipo
        self.dano = dano

    def accion(self):
        print("La TNT está lista para explotar.")

    def colocar(self, orientacion):
        print("TNT colocada en:", orientacion)

    def romper(self):
        print("La TNT se activa y explota.")


class BloqueHorno:
    def __init__(self, color, capacidad_comida):
        self.color = color
        self.capacidad_comida = capacidad_comida

    def accion(self):
        print("Este horno puede cocinar comida.")

    def colocar(self, orientacion):
        print("Horno colocado en:", orientacion)

    def romper(self):
        print("El horno se rompe y suelta los objetos cocinados.")


bloque1 = BloqueCofre(30, 10, "madera")
bloque2 = BloqueCofre(50, 20, "hierro")

bloque3 = BloqueTnt("básica", 100)
bloque4 = BloqueTnt("potente", 200)

bloque5 = BloqueHorno("gris", 5)
bloque6 = BloqueHorno("negro", 10)

bloques = [bloque1, bloque2, bloque3, bloque4, bloque5, bloque6]

for bloque in bloques:
    bloque.accion()
    bloque.colocar("suelo")
    bloque.romper()
    print("------------------------")
